### ISSUE TITLE
Markdown strings shouldn't use their position in the file as keys

### ISSUE THREAD

**Issue by mxmou** [ScratchAddons/website-v2#89]
It means most of the file needs to be translated again after a change.

**Comment by WorldLanguages** [comment]
Yeah, I agree. I believe the website should use string keys, like the extension does.

**Comment by RedGuy12** [comment]
Just enable TX identical string detection smh

I assume there isn't any way of changing the keys inside of MD. the TX setting will see if a key has already been translated before and if it has, users the past translation.
________________________________

**Comment by WorldLanguages** [comment]
lol maybe skip my comment
i didn't read this is about markdown
but i've seen transifex using the string as the key for non-markdown, and i don't think that's very good?

**Comment by mxmou** [comment]
> but i've seen transifex using the string as the key for non-markdown, and i don't think that's very good?

It's much better than using line numbers. Markdown keys were like that too until recently.

**Comment by Hans5958** [comment]
I should have write this a while ago and I forgot, so here it goes now.

This is a Transifex problem on Markdown, and it's really weird that Transifex does that instead of your ordinary hashing or using the string directly. This is a limitation that I didn't expect.

Unless we move providers to something like Crowdin, maybe I should start thinking about doing a YAML/JSON dictionary instead. It builds a hash key from the all the valid strings, and similar strings would have the same key (This may be stupid). It may be worth it in the long run, or not, but I need to consider.

**Comment by Hans5958** [comment]
Forgot to comment it here, but I've since used TM Fillup so you don't need to rewrite exact texts that have been translated. I've tested it on FAQ, which I added admotions, which means the positions are changing, and no translation is thrown.

**Comment by WorldLanguages** [comment]
@Hans5958 That mitigated this issue, good enough I guess
